### DISCUSSION TITLE
bind command: assign Shift-Enter to a command

### DISCUSSION THREAD

**Discussion by sulivanShu** [fish-shell/fish-shell#10069] [Q&A]
Hi!

I would like to assign the `Shift-Enter` key combination to execute the command: "accept-autosuggestion execute". I found how to do it with the Control-o combination:
```
bind \co accept-autosuggestion execute
```
But how to do the same with `Shift-Enter`?

**Comment by faho** [comment]
https://fishshell.com/docs/current/interactive.html#key-sequences

- Run `fish_key_reader`
- Press whatever key you want
- It'll give you something like:

```fish
bind -k enter 'do something'
bind \eOM 'do something'
```

So, use either of the two and replace the "do something" with whatever you want.

**Reply by sulivanShu** [reply]
fish_key_reader specifies `\r` for both `Enter` and `Shift-Enter`. Fish doesn't know the difference between the two?

**Reply by faho** [reply]
Worse: Your *terminal* doesn't differentiate between the two.

`fish_key_reader` accurately displays whatever the terminal sends, and often that simply is the same as something else (for instance tab *is* ctrl-i - these are simply the same sequence).

You would have to either configure your terminal or pick a different one.

Or, as the linked documentation puts it:

> Note that some key combinations are indistinguishable or unbindable. For instance control-i is the same as the tab key. This is a terminal limitation that fish can’t do anything about.
